src/Control/Applicative/Compat.hs

```diff
@@ -19,4 +23,22 @@ instance Monoid a => Monoid (Const a b) where
 instance Monad m => Monad (WrappedMonad m) where
     return = WrapMonad . return
     a >>= f = WrapMonad (unwrapMonad a >>= unwrapMonad . f)
+
+deriving instance Eq a => Eq (ZipList a)
+deriving instance Ord a => Ord (ZipList a)
+deriving instance Read a => Read (ZipList a)
+deriving instance Show a => Show (ZipList a)
+#endif
+
+#if !MIN_VERSION_base(4,8,0)
+deriving instance Eq a => Eq (Const a b)
+deriving instance Ord a => Ord (Const a b)
+
+instance Read a => Read (Const a b) where
+    readsPrec d = readParen (d > 10)
+        $ \r -> [(Const x,t) | ("Const", s) <- lex r, (x, t) <- readsPrec 11 s]
+
+instance Show a => Show (Const a b) where
+    showsPrec d (Const x) = showParen (d > 10) $
+                            showString "Const " . showsPrec 11 x
```

Comment:
Weirdly enough, those definitions are [what `base` currently uses](http://hackage.haskell.org/package/base-4.8.0.0/docs/src/Control-Applicative.html#Const). If they were derived, then `show (Const "a")` would result in `"Const {getConst = \"a\"}"` instead of `"Const \"a\""`.